src/dbus_api/blockdev.rs

```diff
@@ -0,0 +1,285 @@
+                                                       &pool_uuid))
+                        })?;
+    let blockdev_uuid = &blockdev_data.uuid;
+    let blockdev =
+        pool.get_blockdev(blockdev_uuid)
+            .ok_or_else(|| MethodErr::failed(&format!("noblockdev with uuid {}", &blockdev_uuid)))?;
+    i.append(getter(blockdev)?);
+    Ok(())
+}
+
+/// Get the devnode for an object path.
+fn get_blockdev_devnode(i: &mut IterAppend,
+                        p: &PropInfo<MTFn<TData>, TData>)
+                        -> Result<(), MethodErr> {
+    get_blockdev_property(i,
+                          p,
+                          |p| Ok(MessageItem::Str(format!("{}", p.devnode().display()))))
+}
+
+fn get_blockdev_hardware_id(i: &mut IterAppend,
+                            p: &PropInfo<MTFn<TData>, TData>)
+                            -> Result<(), MethodErr> {
+    fn get_hardware_id(blockdev: &BlockDev) -> Result<MessageItem, MethodErr> {
+        match *blockdev.hardware_id() {
+            Some(ref id) => Ok(MessageItem::Str(id.to_owned())),
+            None => Ok(MessageItem::Str("".to_owned())),
+        }
+    }
+
+    get_blockdev_property(i, p, get_hardware_id)
+}
+
+fn get_blockdev_user_id(i: &mut IterAppend,
+                        p: &PropInfo<MTFn<TData>, TData>)
+                        -> Result<(), MethodErr> {
+    fn get_user_id(blockdev: &BlockDev) -> Result<MessageItem, MethodErr> {
+        match *blockdev.user_id() {
+            Some(ref id) => Ok(MessageItem::Str(id.to_owned())),
+            None => Ok(MessageItem::Str("".to_owned())),
+        }
+    }
+
+    get_blockdev_property(i, p, get_user_id)
+}
+
+fn get_blockdev_initialization_time(i: &mut IterAppend,
+                                    p: &PropInfo<MTFn<TData>, TData>)
+                                    -> Result<(), MethodErr> {
+    get_blockdev_property(i,
+                          p,
+                          |p| Ok(MessageItem::Str(p.initialization_time().to_rfc3339())))
+}
+
+fn get_blockdev_physical_size(i: &mut IterAppend,
+                              p: &PropInfo<MTFn<TData>, TData>)
+                              -> Result<(), MethodErr> {
+    get_blockdev_property(i,
+                          p,
+                          |p| Ok(MessageItem::Str(format!("{}", *p.total_size()))))
+}
+
+fn get_blockdev_state(i: &mut IterAppend,
+                      p: &PropInfo<MTFn<TData>, TData>)
+                      -> Result<(), MethodErr> {
+    fn get_state(blockdev: &BlockDev) -> Result<MessageItem, MethodErr> {
+        let state = match blockdev.state() {
+            BlockDevState::Missing => "Missing".into(),
+            BlockDevState::Bad => "Bad".into(),
+            BlockDevState::Spare => "Spare".into(),
+            BlockDevState::NotInUse => "Not-in-use".into(),
+            BlockDevState::InUse => "In-use".into(),
+        };
+        Ok(MessageItem::Str(state))
```

Comment:
Similar to above, would be nice if this were a number, not a string.

Reply:
ok. I'll need to update the api doc.